sdk-workflows/src/main/java/io/dapr/workflows/runtime/WorkflowRuntime.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.workflows.runtime;
+
+import com.microsoft.durabletask.DurableTaskGrpcWorker;
+import com.microsoft.durabletask.DurableTaskGrpcWorkerBuilder;
+import io.dapr.config.Properties;
+
+/**
+ * Contains methods to register workflows and activities.
+ */
+public class WorkflowRuntime implements AutoCloseable {
+
+  private static volatile WorkflowRuntime instance;
+  private DurableTaskGrpcWorkerBuilder builder;
+  private DurableTaskGrpcWorker worker;
+
+  private WorkflowRuntime() throws IllegalStateException {
+
+    if (instance != null) {
+      throw new IllegalStateException("WorkflowRuntime should only be constructed once");
+    }
+
+    int port = Properties.GRPC_PORT.get();
```

Comment:
Can't this use the same client as the main SDK since it invokes the same gRPC stub?